plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php

```diff
@@ -297,14 +297,25 @@ private static function order_received( $order_id = 0 ) {
 			return;
 		}
 
-		$order_customer_id = $order->get_customer_id();
-
-		// For non-guest orders, require the user to be logged in before showing this page.
-		if ( $order_customer_id && get_current_user_id() !== $order_customer_id ) {
-			wc_get_template( 'checkout/order-received.php', array( 'order' => false ) );
-			wc_print_notice( esc_html__( 'Please log in to your account to view this order.', 'woocommerce' ), 'notice' );
-			woocommerce_login_form( array( 'redirect' => $order->get_checkout_order_received_url() ) );
-			return;
+		/**
+		 * Allow to guest users see the page order-received
+		 *
+		 * @param bool $order_received_verification_required If order-received verification is required.
+		 *
+		 * @since 8.4.0
+		 */
+		$order_received_verification_required = apply_filters( 'woocommerce_order_received_verification_required', true );
```

Comment:
What do you think of this slight modification? 

```suggestion
		/**
		 * Indicates if known (non-guest) shoppers need to be logged in before we let
		 * them access the order received page.
		 *
		 * @param bool $verify_known_shoppers If verification is required.
		 *
		 * @since 8.4.0
		 */
		$verify_known_shoppers = apply_filters( 'woocommerce_order_received_verify_known_shoppers', true );
```

Reply:
I think the hook name is a little clearer (otherwise, it's ambiguous as to when it is applied) and it also corrects the description (this does not actually apply to guest users).